### DISCUSSION TITLE
Azure Blob Storage Connectors for Pekko

### DISCUSSION THREAD

**Discussion by Susmit07** [apache/pekko-connectors#893] [General]
Hi Team,

Does pekko have connectors for Azure blob storage, or any plans of releasing in future?

I see one of the fellow engineers (@sfali) have implemented in Alpakka and the PR got merged, so thought of asking 

https://github.com/akka/alpakka/pull/3254/files

Thanks,
Susmit

**Comment by Susmit07** [comment]
@pjfanning  / @raboof 

Tagging the experts for opinion.

**Comment by Susmit07** [comment]
I checked the PR it looks fine by Syed

Shall we fo ahead and implement the same
in Pekko 

I can them work on creating a PR 

What do you guys suggest?

**Reply by pjfanning** [reply]
That is not allowed. Work submitted to the ASF must be your own work. You cannot take someone else's work without their permission. And there is also the risk that the Lightbend team collaborated on the PR and we can not under any circumantances use code written by Lightbend employees (since the date of the Akka license change).

Someone else who has not seen the Akka PR could write a clean room solution.

**Reply by Susmit07** [reply]
I meant taking a reference from the PR its not a copy, actually it would have beem great if @asafali implemented but I will come up
with my own like s3

**Reply by pjfanning** [reply]
If you have read the Alpakka PR, please don't submit any PR to us. You can create your own project and release it yourself. The ASF and the Pekko community cannot accept the risk of being accused of infringing Lightbend's copyright.

**Comment by Susmit07** [comment]
the underlying logic for writing to different storage systems (S3, Azure Blob, etc.) is generally the same, with the main difference being how the SDKs for each platform are implemented. We already have S3 connectors, adding Azure Blob support in a similar manner should be feasible with only minor changes, most of which are related to the specific SDKs you’re using.

It applies for me or for a developer who contributed in Alpakka, we are not reinventing the wheel here

I dont see any reason or copyright infringement
